### PR TITLE
Accept and ignore `using` in method calls (to aid cross-building with Scala 3)

### PR DESCRIPTION
**Pull request by som-snytt**
Users using `using` on Scala 3 may use `using` on Scala 2 in arguments to method invocations, for purposes of cross-compilation. The `using` keyword is ignored and not used in that context under Scala 2.

references https://github.com/lampepfl/dotty/issues/15552

### CODE REVIEW

**Comment by som-snytt** [comment]
`using` ought to be the hit internet karaoke platform. Get it? "you sing"!  Where do I cash in my chips?

File: test/files/neg/i15552.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
+
+class C {
+  val ctx: C = new C()
+  def f() = g(42)(using ctx)  // error: no g
+}
+
+class Innocent {
+  val using = 42
+  def g(i: Int) = i
+  def f() = g(using)
```
Comment (lrytz) [inline test/files/neg/i15552.scala:10]: `g(using using)` works too 😀

**Review by lrytz — APPROVED** [review]

**Comment by lrytz** [comment]
@SethTisue this could go in 2.13.9 I think.

File: test/files/neg/i15552.scala (hunk 1 above)
Comment (som-snytt) [reply test/files/neg/i15552.scala:10]: I wish I had thought of that one. Maybe I'll add another test class `MaliciousPuns`.

**Comment by som-snytt** [comment]
Added the devilish test as suggested, and rebased and squashed.

**Comment by smarter** [comment]
Would you mind opening issues against https://github.com/scalameta/scalameta/issues/ and https://youtrack.jetbrains.com/issues/SCL#newissue to let them know that they need to be update their respective parser for 2.13.9?

**Comment by smarter** [comment]
Also I'd vote for backporting this to 2.12.

**Comment by SethTisue** [comment]
2.12 backport merged: https://github.com/scala/scala/pull/10075

**Comment by SethTisue** [comment]
@som-snytt mind updating the PR description to be user-facing?

**Comment by SethTisue** [comment]
> Would you mind opening issues against [https://github.com/scalameta/scalameta/issues/](https://github.com/scalameta/scalameta/issues/?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc) and [youtrack.jetbrains.com/issues/SCL#newissue](https://youtrack.jetbrains.com/issues/SCL#newissue) to let them know that they need to be update their respective parser for 2.13.9?

The status quo on this is that we use the "release-notes" label to indicate PRs which are compatibility concerns for tooling authors. The label also goes on other sorts of PRs, but the number of PRs with the label isn't that large and thus shouldn't be unwieldy for tooling maintainers to review.

We also use https://contributors.scala-lang.org to announce progress on upcoming releases, and we recently lengthened the span of time (now two weeks) between announcing a release candidate and actually releasing, to give tooling authors more time to adapt to recent changes such as these.

The matter is open to further discussion (on https://contributors.scala-lang.org, please, rather than on this particular PR), but I want to state clearly that the current model is that we expect tooling authors to monitor what's coming from upstream — and we're doing our best to make it reasonably convenient for them to do so. We don't expect our contributors (and ourselves) to notify individual tooling projects of individual changes.

**Comment by smarter** [comment]
I don't know about what should happen in theory, but in practice I've had great success being proactive (https://github.com/scalameta/scalameta/issues?q=is%3Aissue+is%3Aclosed+author%3Asmarter+, https://youtrack.jetbrains.com/issues?q=by:%20smarter%20), whereas here it seems that IntelliJ isn't aware of the change from this PR even though 2.13.9 is about to be released.

Related discussion: https://github.com/scala/improvement-proposals-discussions/discussions/12

**Pull request merged**